### PR TITLE
Make code more general, add legacy repo

### PR DESCRIPTION
**Pull request by friendlyanon**

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -21,12 +20,121 @@
...
+        }))
+    }
+
+
+@gen.coroutine
+def fdroid_parser(package_name: str):
+    template = "https://gitlab.com/fdroid/fdroiddata/raw/master/metadata/{}.yml"
+    url = template.format(package_name)
+    version_data = (yield fetch(url)).body
+    if isinstance(version_data, bytes):
```
Comment (TheAssassin) [inline api.py]: You shouldn't be too short on empty lines. A few more would increase readability.

File: api.py [hunk 2]
```diff
@@ -256,6 +242,9 @@ def update_cache(cls, data):
         now = datetime.now()
         cls._last_request = now
 
+    def data_received(self, chunk):
```
Comment (TheAssassin) [inline api.py]: What's the purpose of this method?

Comment (friendlyanon) [reply api.py]: It's there to silence a warning about unimplemented abstract methods.

Comment (TheAssassin) [reply api.py]: Oh, there's warnings nowadays? Fair enough then.

I haven't touched the code in a while, that's why I'm asking.

**Comment by TheAssassin** [comment]
Thanks! Will redeploy very soon.

File: api.py (hunk 2 above)
Comment (friendlyanon) [reply api.py]: It was just PyCharm complaining, nothing related to the python runtime in particular.

**Comment by friendlyanon** [comment]
I have made a merge request for fdroid with the metadata, I'll notify you when that goes thru, because this API depends on the exitence of every flavor.

**Comment by TobiGr** [comment]
> I have made a merge request for fdroid with the metadata

Thanks!

I think we should announce NewPipe legacy in a blog post once it is available on F-Droid.

**Pull request merged**